editor/src/templates/preview.tsx

```diff
@@ -276,12 +279,28 @@ const initPreview = () => {
           `Error processing the project files: the build result does not contain the preview file: ${previewJSFilePath}`,
         )
       } else {
-        /**
-         * require the js entry point file which will evaluate the module.
-         * the React entry point js file traditionally has a top level side effect,
-         * calling ReactDOM.render() which starts the Preview app.
-         */
-        requireFn('/', previewJSFilePath)
+        try {
+          /**
+           * require the js entry point file which will evaluate the module.
+           * the React entry point js file traditionally has a top level side effect,
+           * calling ReactDOM.render() which starts the Preview app.
+           */
+          requireFn('/', previewJSFilePath)
+        } catch (e) {
+          if (e?.name === ResolvingRemoteDependencyErrorName && e?.message != null) {
+            const loadingMessageDiv = document.createElement('div')
+            loadingMessageDiv.innerHTML = e.message
+            const loadingMessageStyle = [
+              `font-family: -apple-system, BlinkMacSystemFont, Helvetica, 'Segoe UI', Roboto, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol'`,
+              'font-size: 11px',
+              'cursor: default',
+            ].join('; ')
+            loadingMessageDiv.setAttribute('style', loadingMessageStyle)
+            document.body.append(loadingMessageDiv)
```

Comment:
Does this mean this is created for each remote dependency error triggered? Does anything clean them up?

Reply:
It gets cleaned up when the preview is re-rendered, as that completely re-writes the document [here](https://github.com/concrete-utopia/utopia/blob/9ba418253746781432d49cca76c54e4397c1d755/editor/src/templates/preview.tsx#L257)